templates/pre-receive.erb

```diff
@@ -0,0 +1,19 @@
+#! /bin/sh
+while read oldrev newrev refname; do
+  author=$(git log -1 --format='%an' $newrev)
+
+  # superuser always gets a pass
+  if [[ "${author}" == "<%= @admin_username %>" ]]; then
+    continue
+
+  # creating a new branch
+  elif [[ "${oldrev}" == "0000000000000000000000000000000000000000" ]]; then
+    continue
+
+  elif [[ ! $refname =~ ^refs/heads/${author}-? ]]; then
```

Comment:
@binford2k Somewhat of an edge case here, but it's possible for user "greg" to commit to user "gregory"'s branch, I think. If it matters, the regexp could be updated to `^refs/heads/${author}(\-.*?)?$`. Maybe ;)

Reply:
I can check that with Rubular.

Reply:
Not sure how Bash regexps work yet, but do special characters like `/`, `$` and `{` need to be escaped with backslashes?

Reply:
Ok, apparently not.